### PR TITLE
[ADL] Add ISH device to Platform Device list

### PR DESCRIPTION
**Pull request by sindhuragrandhi**
- Add ISH to platform device list.
- Set the ISH address in Post Config phase.

### CODE REVIEW

File: BootloaderCommonPkg/Include/Guid/OsBootOptionGuid.h [hunk 1]
```diff
@@ -72,6 +72,7 @@ typedef enum {
 typedef enum {
   PlatformDeviceMin = OsBootDeviceMax,
   PlatformDeviceGraphics,
+  PlatformDeviceIsh,
```
Comment (gdong1) [inline BootloaderCommonPkg/Include/Guid/OsBootOptionGuid.h:75]: Prefer adding it DeviceTableHobGuid.h.

Comment (sindhuragrandhi) [reply BootloaderCommonPkg/Include/Guid/OsBootOptionGuid.h]: Done

**Comment by bejeanmo** [comment]
Update copyright date in .h files.

**Comment by randylintw** [comment]
UEFI payload can read the value in DeviceTableHob.

**Review by gdong1 — APPROVED** [review]

**Review by bejeanmo — APPROVED** [review]

**Pull request merged**